src/server/page_server.hpp

```diff
@@ -64,13 +64,17 @@ class page_server
     void receive_log_page_fetch (cubpacking::unpacker &upk);
     void receive_data_page_fetch (cubpacking::unpacker &upk);
 
+    void on_log_page_read_result (const LOG_PAGE *log_page, int error_code);
+
     std::unique_ptr<active_tran_server_conn> m_ats_conn;
     std::unique_ptr<active_tran_server_request_queue> m_ats_request_queue;
     std::unique_ptr<active_tran_server_request_autosend> m_ats_request_autosend;
 
     std::unique_ptr<cublog::replicator> m_replicator;
+    std::unique_ptr<cublog::async_page_fetcher> m_log_page_fetcher;
```

Comment:
It is best to have control on the initialization/finalization of the async log page fetcher, because it depends on thread manager. Add functions to initialize/finalize it and they may be called in init_server_type()/finalize_server_type() when server is a page server.